lib/bolt/inventory/group2.rb

```diff
@@ -253,37 +247,30 @@ def data_merge(data1, data2)
         }
       end
 
-      private def fetch_value(data, key, type)
-        value = data.fetch(key, type.new)
-        unless value.is_a?(type)
-          raise ValidationError.new("Expected #{key} to be of type #{type}, not #{value.class}", @name)
-        end
-        value
-      end
-
-      def resolve_aliases(aliases, target_names)
-        @name_or_alias.each do |name_or_alias|
-          # If an alias is found, insert the name into this group. Otherwise use the name as a new target's uri.
-          if target_names.include?(name_or_alias)
-            @targets[name_or_alias] = { 'name' => name_or_alias }
-          elsif (target_name = aliases[name_or_alias])
-            if @targets.include?(target_name)
-              @logger.warn("Ignoring duplicate target in #{@name}: #{target_name}")
+      def resolve_string_targets(aliases, known_targets)
+        @string_targets.each do |string_target|
+          # If this is the name of a target defined elsewhere, then insert the
+          # target into this group as just a name. Otherwise, add a new target
+          # with the string as the URI.
+          if known_targets.include?(string_target)
+            @unresolved_targets[string_target] = { 'name' => string_target }
+          # If this is an alias for an existing target, then add it to this group
+          elsif (canonical_name = aliases[string_target])
+            if local_targets.include?(canonical_name)
+              @logger.warn("Ignoring duplicate target in #{@name}: #{canonical_name}")
             else
-              @targets[target_name] = { 'name' => target_name }
+              @unresolved_targets[canonical_name] = { 'name' => canonical_name }
```

Comment:
Can this go directly into `@resolved_targets`?

Reply:
I think the rationale was that putting it in `@unresolved_targets` ensures it will have all the data keys added to it. That might not matter though, may have just been to quiet a failing test. I'll check it out.

Reply:
Ah I see what the issue was. `data_merge` doesn't gracefully handle missing keys, so it needs to either be added to `@unresolved_targets` or added to `@resolved_targets` with all the missing keys populated.